sci-electronics/kicad/kicad-4.0.0_rc1.ebuild

```diff
@@ -0,0 +1,183 @@
+	x11-libs/cairo"
+DEPEND="${CDEPEND}
+	app-arch/xz-utils
+	doc? ( app-doc/doxygen )
+	i18n? ( >=sys-devel/gettext-0.18 )
+	python? ( dev-lang/swig:0 )"
+RDEPEND="${CDEPEND}
+	sci-electronics/electronics-menu"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+pkg_setup() {
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	if use python; then
+		# dev-python/wxpython doesn't support python3
+		sed '/set(_PYTHON3_VERSIONS 3.3 3.2 3.1 3.0)/d' -i CMakeModules/FindPythonLibs.cmake || die "sed failed"
+	fi
+
+	# Handle optional minimal install.
+	if use minimal; then
+		# remove tempaltes as they are not needed to run binaries
+		sed -e '/add_subdirectory( template )/d' -i CMakeLists.txt || die "sed failed"
+	else
+		# create a link to the parts library in the main project folder
+		ln -s "${WORKDIR}/${P}-library" "${S}/${PN}-library" || die "ln failed"
+		# add the libraries directory to cmake as a subproject to build
+		sed "/add_subdirectory( bitmaps_png )/a add_subdirectory( ${PN}-library )" -i CMakeLists.txt || die "sed failed"
+		# add the libraries directory subproject also to uninstallation procedures
+		sed '/make uninstall/,/# /d' -i ${PN}-library/CMakeLists.txt || die "sed failed"
+	fi
+
+	# Add internationalization for the GUI
+	if use i18n; then
+		# create a link to the translations library in the main project folder
+		ln -s "${WORKDIR}/${P}-i18n" "${S}/${PN}-i18n" || die "ln failed"
+		# Remove unused languages. Project generates only languages specified in the
+		# file in LINGUAS in the subproject folder. By default all languages are added
+		# so we sed out the unused ones based on the user linguas_* settings.
+		local lang=""
+		for lang in ${LANGS}; do
+			if ! use linguas_${lang}; then
+				sed "/${lang}/d" -i ${PN}-i18n/LINGUAS || die "sed failed"
+			fi
+		done
+		# cmakelists does not respect our build dir variables, so make it point to the right location
+		sed "s|\${CMAKE_BINARY_DIR}|${WORKDIR}/${P}_build|g" -i ${PN}-i18n/CMakeLists.txt || die "sed failed"
+		# we also make from the master project so the source dir is understood incorretly, replace that too 
+		sed "s|\${CMAKE_SOURCE_DIR}/\${LANG}|\${CMAKE_SOURCE_DIR}/${PN}-i18n/\${LANG}|g" -i ${PN}-i18n/CMakeLists.txt || die "sed failed"
+		# add the translations directory to cmake as a subproject to build
+		sed "/add_subdirectory( bitmaps_png )/a add_subdirectory( ${PN}-i18n )" -i CMakeLists.txt || die "sed failed"
+		# add the translations directory subproject also to uninstallation procedures
+		sed '/make uninstall/,$d' -i ${PN}-i18n/CMakeLists.txt || die "sed failed"
+	fi
+
+	# Install examples in the right place if requested
+	if use examples; then
+		sed -e 's:${KICAD_DATA}/demos:${KICAD_DOCS}/examples:' -i CMakeLists.txt || die "sed failed"
+	else
+		sed -e '/add_subdirectory( demos )/d' -i CMakeLists.txt || die "sed failed"
+	fi
+
+	# Add important missing doc files
+	sed -e 's/INSTALL.txt/AUTHORS.txt CHANGELOG.txt README.txt TODO.txt/' -i CMakeLists.txt || die "sed failed"
```

Comment:
Really, the amount of `sed` calls ask for tragic failure on version bump. Most of this really belongs in a big build system patch, with some `if (MYNEWOPTION)` conditionals.